Add example for alternate axes scale.

jimmywan opened the pull request:
Fixes altair-viz/altair#1283

jakevdp commented on altair/examples/scatter_alternate_axes_scale.py:
```diff
@@ -0,0 +1,36 @@
+    }
+)
+base_chart =  alt.Chart(chart_df).mark_line().encode(
+    x=alt.X('x', type='quantitative'),
```
change this to ``x='x:Q'``

jakevdp commented on altair/examples/scatter_alternate_axes_scale.py:
```diff
@@ -0,0 +1,36 @@
+"""
+Scatter Plots with alternate Y axis scale
+---------------------------------
```
This should be the same length as the line above it to avoid RST errors when building the docs

jakevdp commented on altair/examples/scatter_alternate_axes_scale.py:
```diff
@@ -0,0 +1,36 @@
+    width=200,
+)
+chart1 = base_chart.encode(
+    y=alt.Y('y', type='quantitative'),
```
``y='y:Q'``

jakevdp commented on altair/examples/scatter_alternate_axes_scale.py:
```diff
@@ -0,0 +1,36 @@
+chart1 = base_chart.encode(
+    y=alt.Y('y', type='quantitative'),
+).properties(title='linear')
+chart2 = base_chart.encode(
```
Let's add a line break between chart definitions for clarity.

jakevdp commented on altair/examples/scatter_alternate_axes_scale.py:
```diff
@@ -0,0 +1,36 @@
+    y=alt.Y('y', type='quantitative'),
+).properties(title='linear')
+chart2 = base_chart.encode(
+    y=alt.Y('y', type='quantitative', scale=alt.Scale(type='log', base=2)),
```
``y=alt.Y('y:Q', scale=...)`` and similar in other places.

jakevdp commented on altair/examples/scatter_alternate_axes_scale.py:
```diff
@@ -0,0 +1,36 @@
+chart4 = base_chart.encode(
+    y=alt.Y('y', type='quantitative', scale=alt.Scale(type='log')),
+).properties(title='log base 10')
+chart1 | chart2 | chart3 | chart4
```
The result here is too wide for the documentation page. Maybe ``(chart1 | chart2) & (chart3 | chart4)`` to make it more compact?

jimmywan commented:
Thanks @jakevdp , I think I've addressed all of your concerns and did a little bit of renaming to make things a bit more explicit.

I'm not really familiar with Sphinx or I'd try to generate the documentation myself. Let me know if there's anything else I can do to improve things.

jakevdp commented:
Great, thanks for the contribution!
